### PR TITLE
changes to get CMEPS to work with both NEMS and CESM

### PR DESCRIPTION
**Pull request by mvertens**
Change Description: new directory structure and files to CMEPS to work with both NEMS and CESM

Testing Completed: testlist_drv.xml - new nuopc baselines created for nov03
    all tests were bit-for-bit except for namelist differences due to cice update
     nuopc scripts_regression_tests.py was not run - since this did not touch any code used by this test

CIME HASH: 0bd9a353e
app_cesm-cmeps HASH: 7aa91e22  
    (updates to cime, clm and cice hashes)

### CODE REVIEW

File: mediator/esmFldsExchange.F90 [hunk 1]
```diff
@@ -2118,7 +2118,7 @@ subroutine esmFldsExchange(gcomp, phase, rc)
     !-----------------------------------------------------------------------------
     ! TODO: add this
     ! if (carma_flds /= ' ') then
-    !    do n = 1,shr_string_listGetNum(carma_flds)
+    !    do n = 1,)number_of_fields in carm_flds)
```
Comment (jedwards4b) [inline mediator/esmFldsExchange.F90]: In comments but wrong as well - remove?

File: mediator/med_phases_prep_atm_mod.F90 [hunk 2]
```diff
@@ -4,6 +4,31 @@ module med_phases_prep_atm_mod
   ! Mediator phases for preparing atm export from mediator
   !-----------------------------------------------------------------------------
 
+  use med_kind_mod          , only : CX=>SHR_KIND_CX, CS=>SHR_KIND_CS, CL=>SHR_KIND_CL, R8=>SHR_KIND_R8
```
Comment (jedwards4b) [inline mediator/med_phases_prep_atm_mod.F90]: why was this done?  It's better inside the subroutine.

File: mediator/med_phases_prep_ice_mod.F90 [hunk 3]
```diff
@@ -4,6 +4,24 @@ module med_phases_prep_ice_mod
   ! Mediator phases for preparing ice export from mediator
   !-----------------------------------------------------------------------------
 
+  use med_kind_mod          , only : CX=>SHR_KIND_CX, CS=>SHR_KIND_CS, CL=>SHR_KIND_CL, R8=>SHR_KIND_R8
```
Comment (jedwards4b) [inline mediator/med_phases_prep_ice_mod.F90]: Why?

File: mediator/med_phases_prep_wav_mod.F90 [hunk 4]
```diff
@@ -4,6 +4,18 @@ module med_phases_prep_wav_mod
   ! Mediator phases for preparing wav export from mediator
   !-----------------------------------------------------------------------------
 
+  use med_kind_mod          , only : CX=>SHR_KIND_CX, CS=>SHR_KIND_CS, CL=>SHR_KIND_CL, R8=>SHR_KIND_R8
```
Comment (jedwards4b) [inline mediator/med_phases_prep_wav_mod.F90]: these don't need to be globals

File: nems/util/shr_abort_mod.F90 [hunk 5]
```diff
@@ -0,0 +1,164 @@
+module shr_abort_mod
+  ! This module defines procedures that can be used to abort the model cleanly in a
```
Comment (jedwards4b) [inline nems/util/shr_abort_mod.F90]: I notice above that you replaced shr_abort_mod with an ESMF abort call - do you still need this?

File: nems/util/shr_kind_mod.F90 [hunk 6]
```diff
@@ -0,0 +1,19 @@
+MODULE shr_kind_mod
```
Comment (jedwards4b) [inline nems/util/shr_kind_mod.F90]: Why is this needed?  Thought you replaced it with med_kinds_mod?

**Comment by mvertens** [comment]
@jedwards4b - its much easier to change things in one place then in
multiple places in the subroutine.

File: mediator/esmFldsExchange.F90 (hunk 1 above)
Comment (mvertens) [reply mediator/esmFldsExchange.F90]: I can remove this - but its a place holder to reference what is needed. I can add a comment as well as opposed to pseudo-code. Would that be better?

File: mediator/med_phases_prep_atm_mod.F90 (hunk 2 above)
Comment (mvertens) [reply mediator/med_phases_prep_atm_mod.F90]: Actually I disagree. For the kind - it should be a module level use statement since it could apply to any subroutine that uses it.

File: mediator/med_phases_prep_ice_mod.F90 (hunk 3 above)
Comment (mvertens) [reply mediator/med_phases_prep_ice_mod.F90]: Same answer as before.

File: mediator/med_phases_prep_wav_mod.F90 (hunk 4 above)
Comment (mvertens) [reply mediator/med_phases_prep_wav_mod.F90]: I disagree.  I believe that the kind references should be global since they only then need to be defined in one place.

File: nems/util/shr_abort_mod.F90 (hunk 5 above)
Comment (mvertens) [reply nems/util/shr_abort_mod.F90]: You still need this for for util - since it specifically references cime/src/share/util routines that use shr_kind_mod. We could rewrite those routines - but I did not take that on right now.

File: nems/util/shr_kind_mod.F90 (hunk 6 above)
Comment (mvertens) [reply nems/util/shr_kind_mod.F90]: see above answer.
